### DISCUSSION TITLE
Read Structure definitions stored in a database

### DISCUSSION THREAD

**Discussion by vivektk84** [FirelyTeam/firely-net-sdk#1775] [Q&A] [answered]
I was exploring a way to move out of file-based usage of profiles and thinking of an approach where I store the profiles into a Database and the IResourceResolver can read those like how it reads from a ZipSource or DirectorySource.

Do we already have a set of APIs available in Firely SDK or can extend at our side so that we can configure the database connection, table, and fields and the ResourceResolver can read it directly from the database instead of a zip file or directory? The intention here is to remove the dependecy with folders/ file systems.

**Comment by marcovisserFurore** [comment, marked as answer]
You can indeed create your own implementation of `IResourceResolver`. There are no helper methods in the SDK to read from databases. But there are enough DB APIs (EntityFramework, ADO.NET)  to read from a database.

**Reply by vivektk84** [reply]
Thanks @marcovisserFurore . EF will be a good option is what i feel.
